lib/aws/rails/sqs_active_job/executor.rb

```diff
@@ -18,11 +19,16 @@ class Executor
 
         def initialize(options = {})
           @executor = Concurrent::ThreadPoolExecutor.new(DEFAULTS.merge(options))
+          # Monitor threads used to refresh visiblity
+          @visibility_refresh = options.delete(:visibility_refresh)
+          @monitor = Concurrent::ThreadPoolExecutor.new(DEFAULTS.merge(options)) if @visibility_refresh
```

Comment:
why do we need another thread here? can't we just use `@executor`, the logic is happening synchronously anyways